.github/workflows/go.yml

```diff
@@ -22,7 +22,21 @@ jobs:
         go-version: ${{ matrix.go }}
 
     - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
```

Comment:
Is it worth investigating how complex it would be to make a checkout action that gets exactly what we need rather than always this static number that will usually be more?

Reply:
It's definitely outside my effective scope. Mainly thinking checkout is written in Typescript and I may spend more time spinning up on that than I would prefer.